src/Amadevus.RecordGenerator.Generators/EqualityUsageAnalyzer.cs

```diff
@@ -0,0 +1,28 @@
+﻿using Amadevus.RecordGenerator.Analyzers;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Amadevus.RecordGenerator.Generators
+{
+    internal sealed class EqualityUsageAnalyzer
+    {
+        public static IEnumerable<Diagnostic> GenerateDiagnostics(RecordDescriptor descriptor)
+        {
+            if (!descriptor.Features.HasFlag(Features.EquatableEquals) 
+             && !descriptor.Features.HasFlag(Features.ObjectEquals)
+             && !descriptor.Features.HasFlag(Features.OperatorEquals)
+             || descriptor.TypeDeclaration.Modifiers.Any(t => t.IsKind(SyntaxKind.SealedKeyword)))
+            {
+                yield break;
+            }
+
+            yield return Diagnostic.Create(
```

Comment:
Lets put this `yield return` inside the `if`.

Then we can add more conditions with more possible diagnostics.

And the next diagnostic needed is that the class cannot have a base class (other than default `object`). I've done it here:

https://github.com/WarHub/wham/blob/af6b15ec1ee8c988e59b4f4223e3bb8673d8a06d/src/WarHub.ArmouryModel.Source.CodeGeneration/Generators/PartialGeneratorBase.cs#L18

> `IsDerived = BaseType.SpecialType != SpecialType.System_Object;`